minetest/minetest
# [no sq] Fix bloom with post_processing_texture_bits < 16 & other stuff

commit 2: as noticed here https://github.com/minetest/minetest/pull/15449#issuecomment-2496141779

explanation for commit 3:
`IShaderSource` was designed with the idea that if you want a shader, you must want it for a node. So it depends heavily on being given a tile material and the node drawtype. But this doesn't make sense neither in theory nor in practice.
This commit takes a small step towards removing this incorrect abstraction.

## To do

This PR is Ready for Review.

## How to test

1. set `post_processing_texture_bits = 8`, enable bloom and confirm that it works
2. test other graphical things
